Library/Formula/stunnel.rb

```diff
@@ -65,9 +66,35 @@ def caveats
         #{etc}/stunnel/stunnel.pem
 
       This certificate will be used by default unless a config file says otherwise!
+      Stunnel will refuse to load the sample configuration file if left unedited.
 
       In your stunnel configuration, specify a SSL certificate with
       the "cert =" option for each service.
     EOS
   end
+
+  test do
+    ENV["HOME"] = testpath
+    (testpath/"tstunnel.conf").write <<-EOS.undent
+      cert = #{etc}/stunnel/stunnel.pem
+
+      setuid = nobody
+      setgid = nobody
+
+      [pop3s]
+      accept  = 995
+      connect = 110
+
+      [imaps]
+      accept  = 993
+      connect = 143
+    EOS
+
+    # Stunnel throws non-zero exit codes on *ANY* failure like a spoilt child, hence
+    # we need to stupidly guard this. Although this is nigh-useless for the bot, the test
+    # allows maintainers and users to manually verify stunnel is behaving as expected.
+    # If words "Configuration successful" appear in `brew test --verbose` then test = success.
+    # The exit 1 error is because to bind ports already in use you tend to require `sudo`.
+    system "(#{bin}/stunnel #{testpath}/tstunnel.conf; true)"
```

Comment:
I tried several different versions of shell output on that and it was still failing hard with exit 1. I even tried piping to an external file and reading that file for the correct words, but still, exit 1 :/.

Sent from my iPhone

> On 4 Jan 2015, at 10:43, Mike McQuaid <notifications@github.com> wrote:
> 
> In Library/Formula/stunnel.rb:
> 
> > +
> > +      [pop3s]
> > +      accept  = 995
> > +      connect = 110
> > +
> > +      [imaps]
> > +      accept  = 993
> > +      connect = 143
> > +    EOS
> > +
> > +    # Stunnel throws non-zero exit codes on *ANY* failure like a spoilt child, hence
> > +    # we need to stupidly guard this. Although this is nigh-useless for the bot, the test
> > +    # allows maintainers and users to manually verify stunnel is behaving as expected.
> > +    # If words "Configuration successful" appear in `brew test --verbose` then test = success.
> > +    # The exit 1 error is because to bind ports already in use you tend to require `sudo`.
> > +    system "(#{bin}/stunnel #{testpath}/tstunnel.conf; true)"
> Let's use shell_output or pipe_output instead and assert on the output.
> 
> —
> Reply to this email directly or view it on GitHub.
>